docs(datasource/custom): rewrite debugging section (#30250)
renovatebot/renovate: lib/modules/datasource/custom/readme.md

### Debugging

Renovate writes tracing logs entries before transformation.
If Renovate finds an unexpected dataformat, it also writes a tracing log _after_ transformation.

#### Getting trace level logs on hosted app

> If you use the Mend Renovate app, use the [`logLevelRemap` config option](https://docs.renovatebot.com/configuration-options/#loglevelremap) to get the trace log.

Author: Please:

- check if this is the correct config option
- suggest the correct link format

> If you use the Mend Renovate app, use the [`logLevelRemap` config option](https://docs.renovatebot.com/configuration-options/#loglevelremap) to get the trace log.

Reviewer: ```suggestion
If you use the Mend Renovate app, use the [`logLevelRemap` config option](../../../self-hosted-configuration.md#loglevelremap) to get the trace log.
```

Author: The published docs have `logLevelRemap` as a _repository_ config option. So `self-hosted-configuration.md` is the wrong file, I think.

I'm skipping this suggestion. But we should still put the right link format in! :wink:

Reviewer: You are right. Copy and paste error of mine.

}
```

#### Getting trace level logs when self-hosting

If you self-host Renovate, follow these steps to get the trace logs:

> 1. Set the [`logFileLevel` config option](https://docs.renovatebot.com/self-hosted-configuration/#logfilelevel) to `trace`

Reviewer: ```suggestion
1. Set the [`logFileLevel` config option](../../../self-hosted-configuration.md#logfilelevel) to `trace`
```

> 1. Run Renovate in `[dryRun` mode](https://docs.renovatebot.com/self-hosted-configuration/#dryrun)

Author: Same thing as above, please check correct name and suggest correct link format.

> 1. Run Renovate in `[dryRun` mode](https://docs.renovatebot.com/self-hosted-configuration/#dryrun)

Reviewer: ```suggestion
1. Run Renovate in [`dryRun` mode](../../../self-hosted-configuration.md#dryrun)
```
